src/hal/src/pso/mod.rs

```diff
@@ -151,26 +151,52 @@ impl fmt::Display for Stage {
 }
 
 /// Shader entry point.
-#[derive(Debug, Copy)]
-pub struct EntryPoint<'a, B: Backend> {
+#[derive(Debug)]
+pub struct EntryPoint<'a, B: Backend, S = RawSpecialization<'a>> {
     /// Entry point name.
     pub entry: &'a str,
     /// Shader module reference.
     pub module: &'a B::ShaderModule,
     /// Specialization.
-    pub specialization: Specialization<'a>,
+    pub specialization: S,
+}
+
+impl<'a, B, S> EntryPoint<'a, B, S>
+where
+    B: Backend,
+    S: Specialization<'a>,
+{
+    /// Convert `specialization` of the `EntryPoint` into raw type
+    /// acceptable by pipeline creation functions.
+    pub fn into_raw(&self, storage: &'a mut SpecializationStorage) -> EntryPoint<'a, B> {
+        EntryPoint {
+            entry: self.entry,
+            module: self.module,
+            specialization: self.specialization.into_raw(storage),
+        }
+    }
 }
 
-impl<'a, B: Backend> Clone for EntryPoint<'a, B> {
+impl<'a, B, S> Clone for EntryPoint<'a, B, S>
```

Comment:
Given that `Backend` already implements `Clone`, we should just put `#[derive(Clone)]` on the struct

Reply:
It does not work that way :(
It requires `B::ShaderModule: Clone` for some reason.

Reply:
I can't reproduce this in playground. This code compiles:
```rust
trait Foo: Clone {
    type Bar;
}

#[derive(Clone)]
struct Tar<'a, F: Foo>(&'a F::Bar);
```

Reply:
But does it implement `Clone`?